### ISSUE TITLE
Incompatibility with the xcolor package when using the scrbook document class

### ISSUE THREAD

**Issue by momeni** [vafa/xepersian#3]
In TeXLive 2016, using the xelatex version _3.14159265-2.6-0.99996_, the **scrbook** version _2016/06/14 v3.21_ document class, the bidi package version _2016/09/12, v19.9_, the xepersian package version _v16.6, 2016/09/09_, and the **xcolor** package version _2016/05/11 v2.12_, prevents compilation.
Compilation error only happens when both the xcolor and the scrbook are used together. First compilation (when the .toc is empty) has no error. Only the second compilation fails.

Minimal example:
```tex
\documentclass[12pt,a4paper]{scrbook}

\usepackage{xcolor}

\usepackage{xepersian}

\settextfont{XB Niloofar}

\title{testing title}
\author{author name}

\begin{document}

\maketitle
\tableofcontents

\clearpage

\chapter{فصل نخست}
\section{بخش}
بخش اول
\subsection{زیربخش}
زیربخش اول

\end{document}
```

**Comment by vafa** [comment]
> ... prevents compilation.

Do you get this on your log?

```latex
\endL or \endR problem (2 missing, 0 extra) detected at line 2
 [][][]\EU1/Yas(0)/m/n/12 ۵[][]

\endL or \endR problem (0 missing, 1 extra) in paragraph at lines 2--2
 [][][] []\EU1/Yas(0)/m/n/12 بخش [][][]  

\endL or \endR problem (2 missing, 0 extra) detected at line 3
 [][][]\EU1/Yas(0)/m/n/12 ۵[][]

\endL or \endR problem (0 missing, 1 extra) in paragraph at lines 3--3
 [][][] []\EU1/Yas(0)/m/n/12 زیربخش [][][]  
) [3] [4]
```
or is it something else?

**Comment by momeni** [comment]
Yes, the error contains the `\endL or \endR problem (2 missing, 0 extra) detected at line 1` message.
Full log file is here: [log.zip](https://github.com/vafa/xepersian/files/469411/log.zip).

**Comment by vafa** [comment]
I released version 20.0 of the `bidi` package and version 16.7 of the `xepersian` package to CTAN which should fix this issue.
